generators/app/templates/src/screens/Dashboard/screens/Home/index.test.tsx

```diff
@@ -0,0 +1,72 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { actionCreators as authActions, ActionTypes, ResetUser } from 'contexts/UserContext/reducer';
+import { withContextProvider as withUserContext } from 'contexts/UserContext';
+import * as AuthService from 'services/AuthService';
+
+import { withContextProvider as withTechContext } from './context';
+
+import Home from '.';
+
+describe('Home component', () => {
+  afterAll(() => {
+    jest.clearAllMocks();
+  });
+
+  test('logouts when clicks logout button', async () => {
+    const logout = jest.fn();
+    const resetUser: () => ResetUser = jest.fn(() => ({ type: ActionTypes.RESET_USER }));
+    const removeCurrentUserToken = jest.fn();
+    jest.spyOn(AuthService, 'logout').mockImplementationOnce(logout);
+    jest.spyOn(authActions, 'resetUser').mockImplementationOnce(resetUser);
+    jest.spyOn(AuthService, 'removeCurrentUserToken').mockImplementationOnce(removeCurrentUserToken);
+
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    const RenderHOC = withUserContext(Home);
+    render(<RenderHOC />);
+
+    const logoutButton = screen.getByRole('button', { name: /Home:logout/ });
+    userEvent.click(logoutButton);
+
+    await waitFor(() => expect(logout).toHaveBeenCalled());
```

Comment:
is this waitfor necessary? If it is, shouldn't we waitFor the click, which I guess is the code that could take some time? If you keep the waitFor and move it to userEvent.click, change the others so all waitFors do the same

Reply:
the click is synchronous but will trigger some changes in the component, that changes are waited by `waitFor` that is why is necessary.